### PR TITLE
[BEAM-7389] Add code examples for Regex page

### PR DESCRIPTION
**Pull request by davidcavazos**
Adding code samples for the `Regex` page.

R: @aaltay @rosetn Can you take a look at this whenever you have a chance? Thanks!

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by davidcavazos** [comment]
Staged: http://apache-beam-website-pull-requests.storage.googleapis.com/9262/documentation/transforms/python/elementwise/regex/index.html

File: website/src/documentation/transforms/python/element-wise/regex.md [hunk 1]
```diff
@@ -19,10 +19,151 @@ limitations under the License.
...
-See [BEAM-7389](https://issues.apache.org/jira/browse/BEAM-7389) for updates. 
 
-## Related transforms 
-* [Map]({{ site.baseurl }}/documentation/transforms/python/elementwise/map) applies a simple 1-to-1 mapping function over each element in the collection
\ No newline at end of file
+In the following examples, we create a pipeline with a `PCollection` of text strings.
+Then, we use the `re` module to search, replace, and split through the text elements using
+[regular expressions](https://docs.python.org/3/library/re.html).
+
+You can use tools to help you create and test your regular expressions such as
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: Add comma

"expressions such as"->"expressions, such as"

File: website/src/documentation/transforms/python/element-wise/regex.md [hunk 2]
```diff
@@ -19,10 +19,151 @@ limitations under the License.
...
+[regex101](https://regex101.com/),
+make sure to specify the Python flavor at the left side bar.
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: Split into a new sentence.

"Make sure to specify the Python flavor at the left side bar."

File: website/src/documentation/transforms/python/element-wise/regex.md [hunk 3]
```diff
@@ -19,10 +19,151 @@ limitations under the License.
...
+
+### Example 1: Regex match
+
+[`re.match`](https://docs.python.org/3/library/re.html#re.match)
+will try to match the regular expression from the beginning of the string.
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: "will try to"->"tries to"

File: website/src/documentation/transforms/python/element-wise/regex.md [hunk 4]
```diff
@@ -19,10 +19,151 @@ limitations under the License.
...
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Example 2: Regex search
+
+[`re.search`](https://docs.python.org/3/library/re.html#re.search)
+will try to search for the first occurrence the regular expression in the string.
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: "will try to search"->"tries to search"

File: website/src/documentation/transforms/python/element-wise/regex.md [hunk 5]
```diff
@@ -19,10 +19,151 @@ limitations under the License.
...
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Example 3: Regex find all
+
+[`re.finditer`](https://docs.python.org/3/library/re.html#re.finditer)
+will try to search for all the occurrence the regular expression in the string.
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: "will try to search"->"tries to search"

File: website/src/documentation/transforms/python/element-wise/regex.md [hunk 6]
```diff
@@ -19,10 +19,151 @@ limitations under the License.
...
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Example 5: Regex split
+
+[`re.split`](https://docs.python.org/3/library/re.html#re.split)
+will split the string on every ocurrence of the regular expression.
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: "will split"->"splits"

File: website/src/documentation/transforms/python/element-wise/regex.md [hunk 7]
```diff
@@ -19,10 +19,151 @@ limitations under the License.
...
+      View on GitHub
+    </a>
+  </td>
+</table>
+<br>
+
+### Example 4: Regex replace
+
+[`re.sub`](https://docs.python.org/3/library/re.html#re.sub)
+will substitute occurrences the regular expression in the string.
```
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: "will substitute"->"substitutes"

Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: "occurrences the"->"occurrences of the"

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 5 above)
Comment (rosetn) [inline website/src/documentation/transforms/python/element-wise/regex.md]: "for all the"->"for all of the"

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 1 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Added

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 2 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 3 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 4 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 5 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 7 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

File: website/src/documentation/transforms/python/element-wise/regex.md (hunk 6 above)
Comment (davidcavazos) [reply website/src/documentation/transforms/python/element-wise/regex.md]: Done

**Comment by davidcavazos** [comment]
@aaltay @rosetn This is ready for review, note that the last code snippet had a typo on the closing region tag, but it's fixed here too.

**Comment by aaltay** [comment]
I see an extra space in the example data:
`'🍓,   Strawberry,   perennial',`

do you know why? (between the strawberry icon and strawberry text.)

**Comment by davidcavazos** [comment]
It's to show that `' *'` can match zero or *more* spaces. If you think it's confusing I can make them all the same.

**Comment by aaltay** [comment]
I think it is confusing. I will rather fix that and assume that people understand basics of regex patterns.

**Comment by davidcavazos** [comment]
@aaltay Normalized whitespaces in regex examples.

Also, the [Pydoc](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.util.html#apache_beam.transforms.util.Regex) is not showing even if the code is already released. I'm still adding the link so whenever it's ready it should work, so it shouldn't be a showstopper for this. Is there any way to regenerate and publishing the Pydocs or should we just wait until the next release?

**Comment by davidcavazos** [comment]
Run Portable_Python PreCommit

**Comment by aaltay** [comment]
We can wait until the next release for pydocs. Their main use is for released versions really.

**Pull request merged**